### PR TITLE
feat: Contact Form

### LINKED ISSUE #63: [Feature] Complete contact form
**Is your feature request related to a problem? Please describe.**
Contact form currently doesn't not function

**Describe the solution you'd like**
Complete the functionality of the contact form.
This could be done using a static form service such as web3forms or through the backend possibly with emailjs or us-abase.

**Describe alternatives you've considered**
Deleting the contact form as mailto currently still exist.

**Additional context**
This functionality was left incomplete the previous group

### PR DESCRIPTION
**Pull request by cste303**
Made the contact form functional and moved it to its own component.

### Changes

Describe the Pull Request here. Add any references and info to help reviewers understand your changes.

Made the contact form functional through Web3Forms. It sends the email to the degreeplanner90@gmail.com account. Also moved the contact form into its own component.

- Functional contact form
- Contact form its own component

### Tests applied

What testing has been applied to validate changes are accurate and correct

- manual testing

### Issue ticket number(s)

Enter the issue numbers resolved by this pull request

- closes #63 

### Checklist

- [x] Documented
- [x] Linting tests successful
- [x] merge updated prod branch into development branch

### CODE REVIEW

File: components/ContactForm.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  };
+
+  return (
+    <div className="container mx-auto">
+      <div className="m-7">
+        <form
+          className="flex-1 flex flex-col mx-auto w-full max-w-md justify-center text-left gap-2 text-foreground"
+          onSubmit={handleFormSubmit}
+        >
+          <input type="hidden" name="access_key" value="8347a957-fbc8-4284-9c75-27c5dbe46292" />
```
Comment (hwan513) [inline components/ContactForm.tsx:48]: Maybe we could reference web3forms here. Also perhaps we could add some comment to indicate which email this contact form is being sent to (maybe make it the gmail one we have).

File: components/ContactForm.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          <input type="hidden" name="subject" value="New Submission from Web3Forms" />
```
Comment (hwan513) [inline components/ContactForm.tsx]: We could probably change this value="New submission from Degree Planner.

File: components/ContactForm.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+          <label className="text-md" htmlFor="message">
+            Description
+          </label>
+          <textarea
+            id="message"
+            className="rounded-md px-4 py-2 bg-inherit border mb-6"
+            name="Message"
+            placeholder="Enter your message..."
+            rows={4}
+            style={{ minHeight: '8rem' }}
```
Comment (hwan513) [inline components/ContactForm.tsx]: Style can be moved into the tailwind className with `min-h-[8rem]`.

File: components/ContactForm.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+          onSubmit={handleFormSubmit}
+        >
+          <input type="hidden" name="access_key" value="8347a957-fbc8-4284-9c75-27c5dbe46292" />
+          <input type="hidden" name="subject" value="New Submission from Web3Forms" />
+          <input type="checkbox" name="botcheck" style={{ display: 'none' }} />
+
+          <label className="text-md" htmlFor="email">
+            Email
+          </label>
+          <input
```
Comment (hwan513) [inline components/ContactForm.tsx:55]: Our input fields could probably all be required.

**Review by hwan513 — CHANGES_REQUESTED** [review]
Changes look good. Just have a few suggestions.

File: components/ContactForm.tsx (hunk 4 above)
Comment (cste303) [reply components/ContactForm.tsx:55]: Yeah good idea will add.

File: components/ContactForm.tsx (hunk 3 above)
Comment (cste303) [reply components/ContactForm.tsx]: Good spotting will do

File: components/ContactForm.tsx (hunk 2 above)
Comment (cste303) [reply components/ContactForm.tsx]: Ahh yeah will do that

File: components/ContactForm.tsx (hunk 1 above)
Comment (cste303) [reply components/ContactForm.tsx:48]: Yeah will chuck that in

**Review by hwan513 — APPROVED** [review]
Looks good to me 👍

**Pull request merged**